test/common/utils/lodash-replacement.unit.spec.js

```diff
@@ -0,0 +1,290 @@
+      const obj = {
+        a: 1,
+        b: 2
+      };
+      const cloned = _.clone(obj);
+      expect(cloned).to.eql(obj);
+      expect(cloned).to.not.equal(obj);
+    });
+
+    it('should clone a set', () => {
+      const s = new Set([{ sub: 'original' }]);
+      const cloned = _.clone(s);
+      expect(cloned).to.eql(s);
+      expect(cloned).to.not.equal(s);
+    });
+  });
+
+
+  describe('cloneDeep', () => {
+    const obj = {
+      'int': 1,
+      obj: {
+        foo: 'bar',
+        nestedObj: {
+          nestedArray: [0, 2, 4]
+        }
+      },
+      string: 'string!',
+      array: ['0', 1, null, { s: 'thing', o: { k: 'I am nested' } }],
+      func: function() { return this.int; }, // eslint-disable-line
+      arrowFunction: () => this.int
+    };
+
+    const cloned = _.cloneDeep(obj);
+
+    it('should return an object with a different reference', () => {
+      expect(cloned).to.eql(obj);
+      expect(cloned).to.not.equal(obj);
+    });
+
+    it('should handle an array as the root "object"', () => {
+      const arr = ['foo', 'bar', 0];
+      const clonedArr = _.cloneDeep(arr);
+
+      expect(clonedArr).to.eql(arr);
+      expect(clonedArr).to.not.equal(arr);
+    });
+
+    it('should clone all sub-objects', () => {
+      expect(cloned.obj).to.eql(obj.obj);
+      expect(cloned.obj).to.not.equal(obj.obj);
+    });
+
+    it('should clone all sub-arrays', () => {
+      expect(cloned.array).to.eql(obj.array);
+      expect(cloned.array).to.not.equal(obj.array);
+      expect(cloned.obj.nestedObj.nestedArray).to.eql(obj.obj.nestedObj.nestedArray);
+      expect(cloned.obj.nestedObj.nestedArray).to.not.equal(obj.obj.nestedObj.nestedArray);
+    });
+
+    it('should clone all objects in arrays', () => {
+      expect(cloned.array[3]).to.eql(obj.array[3]);
+      expect(cloned.array[3]).to.not.equal(obj.array[3]);
+    });
+  });
+
+
+  describe('get', () => {
+    const o = {
+      a: 'a',
+      b: { c: 'c' },
+      k: { a: { y: 'f' } },
+      g: ['h', 'i', 'j']
+    };
+
+    it('should get a value one level deep', () => {
+      expect(_.get('a', o)).to.equal(o.a);
+    });
+
+    it('should get a value n levels deep', () => {
+      expect(_.get('b.c', o)).to.equal(o.b.c);
+      expect(_.get('k.a.y', o)).to.equal(o.k.a.y);
+    });
+
+    it('should handle array indexes', () => {
+      expect(_.get('g[2]', o)).to.equal(o.g[2]);
+    });
+
+    it('should handle an array path', () => {
+      expect(_.get(['k', 'a', 'y'], o)).to.equal(o.k.a.y);
+    });
+
+    it('should return a default value if not found', () => {
+      expect(_.get('does.not.exist', o, 'default')).to.equal('default');
+    });
+
+    it('should return undefined if not found and no default is provided', () => {
+      expect(_.get('does.not.exist', o)).to.equal();
+    });
+  });
+
+
+  describe('set', () => {
+    it('should set the value of an existing path', () => {
+      const o = {
```

Comment:
It looks like this object is the same for all the tests, so you could define it once in the describe block.

Reply:
I was going to, but since we're also testing whether this is mutative on the original object, I thought it would be safer to create a new object per test.